### PR TITLE
Voters can add options

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Voters can add options if allowed by the poll creator.

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Scripts/Controllers/BasicVoteController.js [hunk 1]
```diff
@@ -53,6 +61,15 @@
 
         }
 
+        function addOption() {
```
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/BasicVoteController.js]: We might want to factor this out to the parent Vote controller, as it seems to be identical in each class

Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Scripts/Controllers/BasicVoteController.js]: The next set of changes I'm making are looking at refactoring as much as possible out of the *VoteControllers and into the VotingPageController.

**Comment by tpkelly** [comment]
:+1:

File: VotingApplication/VotingApplication.Web/Api/Controllers/PollVoteController.cs [hunk 2]
```diff
@@ -1,93 +1,180 @@
...
+                foreach (VoteRequestModel voteRequest in voteRequests)
+                {
+                    Option option = context
+                        .Options
+                        .Single(o => o.Id == voteRequest.OptionId);
+
+                    Vote modelToVote = ModelToVote(voteRequest, ballot, option, poll);
+                    context.Votes.Add(modelToVote);
+
+                    // TODO: refactor the voteRequest model to be a ballotRequest instead. => only one voterName.
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Api/Controllers/PollVoteController.cs]: Might have missed it, but is there a JIRA Issue for this? It's been a TODO for a while now.

**Comment by stevenhillcox** [comment]
Again, happy to pull when this merges

**Pull request merged**